dali/pipeline/operator/arg_helper_test.cc

```diff
@@ -60,41 +61,104 @@ void ArgValueTestTensorInput(TensorListShape<ndim> ts) {
   }
 }
 
+template <int ndim, typename... AcquireArgs>
+void ArgValueTestAllowEmpty(TensorListShape<ndim> expected_sh, AcquireArgs... args) {
+  ASSERT_TRUE(is_uniform(expected_sh));  // the test assumes that
+  auto sh = expected_sh;
+  sh.tensor_shape_span(2)[0] = 0;  // this makes the sample with idx 2 empty
+
+  EXPECT_THROW(ArgValueTestTensorInput<ndim>(sh, expected_sh, ArgValue_Default),
+               std::runtime_error);
+  ArgValueTestTensorInput<ndim>(sh, expected_sh, ArgValue_AllowEmpty);
+
+  auto expected_sample_sh = sh[0];
+  EXPECT_THROW(ArgValueTestTensorInput<ndim>(sh, expected_sample_sh, ArgValue_Default),
+               std::runtime_error);
+  ArgValueTestTensorInput<ndim>(sh, expected_sample_sh, ArgValue_AllowEmpty);
+
+  OpSpec spec("MTTransformAttr");  // need to use a real op name
+  ArgumentWorkspace ws;
+  auto arg_data = std::make_shared<TensorVector<CPUBackend>>();
+  SetupData(*arg_data, sh);
+  ws.AddArgumentInput("M", arg_data);
+  spec.AddArgumentInput("M", "M");
+  ArgValue<float, ndim> arg("M", spec);
+  arg.Acquire(spec, ws, kNumSamples, expected_sample_sh, ArgValue_AllowEmpty);
+
+  EXPECT_FALSE(arg.IsEmpty(0));
+  EXPECT_FALSE(arg.IsEmpty(1));
+  EXPECT_TRUE(arg.IsEmpty(2));
```

Comment:
Can you tell my why 2nd arg is empty? `SetupData` fills them all.

Reply:
Because the shape is empty:
```sh.tensor_shape_span(2)[0] = 0;  // this makes the sample with idx 2 empty```

Reply:
Understood.
I'm not sure is the silent assumption that `expected_sh` has the necessary length in L68, and the arg has  5 elements reads well in this test.
If one changes `kNumSamples` failures here may be a bit surprising.

Reply:
How about now? I changed it to a loop